### PR TITLE
Add docstring and simplify signature `newton_system`

### PR DESCRIPTION
**Pull request by tmigot**
The next step will refactor the core of the function slightly. This should help debug and update the usage of the LDL solver. #43

### CODE REVIEW

File: src/CaNNOLeS.jl [hunk 1]
```diff
@@ -623,15 +623,34 @@ function SolverCore.solve!(
...
 end
 
-function newton_system(x, r, λ, Fx, rhs, LDLT, ρold, params, method, linsolve)
-  nvar = length(x)
-  nequ = length(r)
-  ncon = length(λ)
+"""
+    newton_system(nvar, nequ, ncon, rhs, LDLT, ρold, params, linsolve)
+
+Compute an LDLt factorization of the (`nvar + nequ + ncon`)-square matrix for the Newton system contained in `LDLT` with the method `linsolve`, i.e. `sparse(LDLT.rows, LDLT.cols, LDLT.vals, N, N)`.
```
Comment (dpo) [inline src/CaNNOLeS.jl:629]:
```suggestion
Compute an LDLt factorization of the (`nvar + nequ + ncon`)-square matrix for the Newton system contained in `LDLT`, i.e., `sparse(LDLT.rows, LDLT.cols, LDLT.vals, N, N)`, with the method `linsolve`.
```

File: src/CaNNOLeS.jl [hunk 2]
```diff
@@ -623,15 +623,34 @@ function SolverCore.solve!(
   return stats
 end
 
-function newton_system(x, r, λ, Fx, rhs, LDLT, ρold, params, method, linsolve)
```
Comment (dpo) [inline src/CaNNOLeS.jl:626]: Maybe leave a function with the old signature that calls the new one, and mark it as deprecated?

File: src/CaNNOLeS.jl [hunk 3]
```diff
@@ -623,15 +623,34 @@ function SolverCore.solve!(
...
+    newton_system(nvar, nequ, ncon, rhs, LDLT, ρold, params, linsolve)
+
+Compute an LDLt factorization of the (`nvar + nequ + ncon`)-square matrix for the Newton system contained in `LDLT` with the method `linsolve`, i.e. `sparse(LDLT.rows, LDLT.cols, LDLT.vals, N, N)`.
+If the factorization fails, a new factorization is attempted with an increased value for the regularization ρ as long as it is smaller than `params[:ρmax]`.
+The factorization is then used to solve the linear system whose right-hand side is `rhs`.
+
+# Output
+
+- `d`: the solution of the linear system;
+- `ρ`: the last value of the regularization parameter;
```
Comment (dpo) [inline src/CaNNOLeS.jl:636]:
```suggestion
- `ρ`: the value of the regularization parameter used in the factorization;
```

File: src/CaNNOLeS.jl [hunk 4]
```diff
@@ -623,15 +623,34 @@ function SolverCore.solve!(
...
+- `ρold`: the old regularization value. It is equal to `ρ` if `ρ <= params[:ρmax]`;
```
Comment (dpo) [inline src/CaNNOLeS.jl:637]: This is not very clear. What is the "old" value?

File: src/CaNNOLeS.jl (hunk 2 above)
Comment (tmigot) [reply src/CaNNOLeS.jl:626]: Thanks for the comments. This is added.

File: src/CaNNOLeS.jl (hunk 4 above)
Comment (tmigot) [reply src/CaNNOLeS.jl]: I tried to clarify this as: the value of the regularization parameter used in the previous successful factorization, or 0 if this is the first one;

**Pull request merged**